Quadriga/src/main/java/edu/asu/spring/quadriga/utilities/impl/TextXMLParser.java

```diff
@@ -0,0 +1,93 @@
+package edu.asu.spring.quadriga.utilities.impl;
+
+import java.io.IOException;
+import java.io.StringReader;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import edu.asu.spring.quadriga.dao.impl.textfile.TextFileDAO;
+import edu.asu.spring.quadriga.domain.factory.workspace.ITextFileFactory;
+import edu.asu.spring.quadriga.domain.workspace.ITextFile;
+import edu.asu.spring.quadriga.exceptions.TextFileParseException;
+import edu.asu.spring.quadriga.utilities.ITextXMLParser;
+
+/**
+ * Implementation class for ITextXMLParser
+ * 
+ * @author Nischal Samji
+ *
+ */
+@Service
+public class TextXMLParser implements ITextXMLParser {
+
+    @Autowired
+    private ITextFileFactory txtFileFactory;
+
+    private static final Logger logger = LoggerFactory.getLogger(TextXMLParser.class);
+    
+    @Override
+    public ITextFile parseTextXML(String xml, String wsId, String projId) throws TextFileParseException {
+
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+
+        DocumentBuilder builder;
+        Document document = null;
+        try {
+            builder = factory.newDocumentBuilder();
+            document = builder.parse(new InputSource(new StringReader(xml)));
+        } catch (ParserConfigurationException | IOException e) {
+            logger.error("Error while parsing the XML", e);
```

Comment:
don't log the whole stack trace here. You can log a message but assume that the exception will be handled above (meaning someone will do the correct thing when an exception is thrown which also includes logging). Otherwise, there will be two stacktraces printed for the same exception. Does this make sense?